bot/cogs/trivia.py

```diff
@@ -0,0 +1,307 @@
+        leaders.sort(key=lambda x: x[1])
+        leaders.reverse()
+        description = ""
+        for count, leader in enumerate(leaders):
+            if count == 0:
+                description += f"🥇: {leader[0]} with {leader[1]} points\n\n"
+            elif count == 1:
+                description += f"🥈: {leader[0]} with {leader[1]} points\n\n"
+            elif count == 2:
+                description += f"🥉: {leader[0]} with {leader[1]} points\n\n"
+            elif count == 25:
+                break
+            else:
+                description += f"{count+1}: {leader[0]} with {leader[1]} points\n"
+        if not description:
+            description = "No one has played yet!"
+        embed = Embed(
+            title="Trivia top 25 leaderboard",
+            description=description,
+            color=EMBED_COLOR
+        )
+        if str(ctx.author.id) in self.scores:
+            embed.set_footer(text=f"{ctx.author.name} you have {self.scores[str(ctx.author.id)][1]} points")
+        await ctx.send(embed=embed)
+
+    async def get_trivia(self, category: str) -> dict:
+        """Function Used to generate the dictionary with question data."""
+        if not category:
+            url = "https://opentdb.com/api.php?amount=1&type=multiple&token="
+        else:
+            url = f"https://opentdb.com/api.php?amount=1&category={category}&type=multiple&token="
+        reset_url = "https://opentdb.com/api_token.php?command=reset&token="
+        while True:
+            async with self.bot.session.get(url+self.TOKEN) as resp:
+                data = await resp.json()
+            if data['response_code'] == 0:
+                break
+            elif data['response_code'] == 3:
+                async with self.bot.session.get('https://opentdb.com/api_token.php?command=request') as resp:
+                    self.TOKEN = (await resp.json())['token']
+                    print('Generating Token')
+            elif data['response_code'] == 4:
+                await self.bot.session.get(reset_url+self.TOKEN)
+        data = data['results'][0]
+
+        answers = [
+            unescape(data['incorrect_answers'][0]),
+            unescape(data['incorrect_answers'][1]),
+            unescape(data['incorrect_answers'][2]),
+            unescape(data['correct_answer']),
+        ]
+        shuffle(answers)
+        for _count, item in enumerate(answers):
+            if item == data['correct_answer']:
+                break
+
+        if data['difficulty'] == 'easy':
+            points = 5
+        elif data['difficulty'] == 'medium':
+            points = 10
+        elif data['difficulty'] == 'hard':
+            points = 20
+
+        return {
+            'correct': _count+1,
+            'answers': answers,
+            'difficulty': data['difficulty'],
+            'question': unescape(data['question']),
+            'points': points,
+            'category': f"Category: {data['category']}"
+        }
```

Comment:
this could use a dictionary as well to get the points based on difficulty i.e `my_dict = {"easy": 5}`
this will also fix the issue where points may not be defined, as you could do `return { "points": points_table[data["difficulty"]]}` (and difficulty can be it's own var since it is being used more than once)